docs/sanitation/sanitations.bal

```diff
@@ -0,0 +1,430 @@
+        boolean schemaNameCheck = schemaName.includes(".");
+        if schemaNameCheck {
+            string updatedKey = getSanitizedSchemaName(schemaName);
+            updatedSchemas[updatedKey] = schema;
+            updatedNames[schemaName] = updatedKey;
+        } else {
+            updatedSchemas[schemaName] = schema;
+        }
+    }
+    spec.components.schemas = updatedSchemas;
+
+    string updatedSpec = spec.toJsonString();
+    foreach [string, string] [oldName, newName] in updatedNames.entries() {
+        string sanitizedOldNameRegex = re `\.`.replace(oldName, "\\.");
+        regexp:RegExp regexp = re `${sanitizedOldNameRegex}"`;
+        updatedSpec = regexp.replaceAll(updatedSpec, newName + "\"");
+    }
+
+    check io:fileWriteString(specPath, updatedSpec);
+}
+
+function getSanitizedSchemaName(string schemaName) returns string {
+    int? indexOfPeriod = schemaName.lastIndexOf(".");
+    int substringStartIndex = indexOfPeriod == () ? 0 : indexOfPeriod + 1;
+    string updatedKey = schemaName.substring(substringStartIndex);
+
+    if updatedKey.endsWith("_Type") {
+        updatedKey = updatedKey.substring(0, updatedKey.length() - 5);
+    }
+
+    if updatedKey.endsWith("_Type-create") {
+        updatedKey = "Create" + updatedKey.substring(0, updatedKey.length() - 12);
+    }
+
+    if updatedKey.endsWith("_Type-update") {
+        updatedKey = "Update" + updatedKey.substring(0, updatedKey.length() - 12);
+    }
+
+    if updatedKey.endsWith("Type") {
+        updatedKey = updatedKey.substring(0, updatedKey.length() - 4);
+    }
+
+    if updatedKey.endsWith("Type-create") {
+        updatedKey = "Create" + updatedKey.substring(0, updatedKey.length() - 11);
+    }
+
+    if updatedKey.endsWith("Type-update") {
+        updatedKey = "Update" + updatedKey.substring(0, updatedKey.length() - 11);
+    }
+
+    if updatedKey.endsWith("-create") {
+        updatedKey = "Create" + updatedKey.substring(0, updatedKey.length() - 7);
+    }
+
+    if updatedKey.endsWith("-update") {
+        updatedKey = "Update" + updatedKey.substring(0, updatedKey.length() - 7);
+    }
+    return updatedKey;
+}
+
+function sanitizResponseSchemaNames(string specPath) returns error? {
+    json openAPISpec = check io:fileReadJson(specPath);
+    Specification spec = check openAPISpec.cloneWithType(Specification);
+    boolean isODATA4 = false;
+    if (spec.x\-sap\-api\-type=="ODATAV4"){
+        isODATA4 = true;
+    }
+    map<Path> paths = spec.paths;
+    foreach var [key, value] in paths.entries() {
+        if value.get != () {
+            Get getPath = value.get ?: {parameters: []};
+            map<ResponseCode> responses = getPath.responses ?: {};
+            foreach [string,ResponseCode] [_,item] in responses.entries() {
+                if item.description == "Retrieved entities" {
+                    map<ResponseHeader> content = item.content ?: {};
+                    if content.hasKey("application/json"){
```

Comment:
Is this check necessary. Since you are assigning default values if the value is null, no need to check beforehand